invenio/modules/search/testsuite/test_elasticsearch_walker.py

```diff
@@ -0,0 +1,321 @@
+                                }
+                            ]
+                        }
+                    }
+                ]
+            }
+        })
+
+    def test_combined_bool(self):
+        tree = OrOp(AndOp(AndOp(ValueQuery(Value('aaa')),
+                                ValueQuery(Value('bbb'))),
+                          NotOp(ValueQuery(Value('ccc')))),
+                    ValueQuery(Value('ddd')))
+        ll = {
+                "multi_match": {
+                    "fields": ["_all"],
+                    "query": "ddd"
+                }
+             }
+        rr = {
+                "bool": {
+                    "must": [
+                        {
+                            "multi_match": {
+                                "fields": ["_all"],
+                                "query": "aaa"
+                            }
+                        },
+                        {
+                            "multi_match": {
+                                "fields": ["_all"],
+                                "query": "bbb"
+                            }
+                        }
+                    ]
+                }
+             }
+        rl = {
+                "bool": {
+                    "must_not": [
+                        {
+                            "multi_match": {
+                                "fields": ["_all"],
+                                "query": "ccc"
+                            }
+                        }
+                    ]
+                }
+             }
+        r = {
+                "bool": {
+                    "must": [rr, rl]
+                }
+            }
+        d = {
+                "bool": {
+                    "should": [r, ll]
+                }
+            }
+        self.assertEqual(tree.accept(self.converter), d)
+
+    # Spires-syntax queries
+    def test_simple_spires(self):
+        tree = SpiresOp(Keyword('t'), Value('quark'))
+        self.assertEqual(tree.accept(self.converter), {
+            "multi_match": {
+                "fields": ["t"],
+                "query": "quark"
+            }
+        })
+
+    def test_combined_spires(self):
+        tree = AndOp(SpiresOp(Keyword('t'), Value('quark')),
+                     SpiresOp(Keyword('a'), Value('ellis')))
+        self.assertEqual(tree.accept(self.converter), {
+            "bool": {
+                "must": [
+                    {
+                        "multi_match": {
+                            "fields": ["t"],
+                            "query": "quark",
+                        }
+                    },
+                    {
+                        "multi_match": {
+                            "fields": ["a"],
+                            "query": "ellis",
+                        }
+                    }
+
+                ]
+            }
+        })
+
+    # Operators
+    def test_greater_op(self):
+        tree = SpiresOp(Keyword('date'), GreaterOp(Value('1984')))
```

Comment:
@jirikuncar 
This is how the invenio-query-parser works. 
Check the [tests](https://github.com/inveniosoftware/invenio-query-parser/blob/master/tests/test_parser.py#L269)